src/dos/cdrom_ioctl_linux.cpp

```diff
@@ -53,12 +57,169 @@ bool CDROM_Interface_Ioctl::GetUPC(unsigned char &attr, char *upc)
 	return (ret > 0);
 }
 
-bool CDROM_Interface_Ioctl::ReadSectors(PhysPt buffer,
-                                        const bool raw,
-                                        const uint32_t sector,
-                                        const uint16_t num)
+bool CDROM_Interface_Ioctl::GetAudioTracks(uint8_t& stTrack, uint8_t& end, TMSF& leadOut)
+{
+	if (cdrom_fd == -1) {
+#ifdef DEBUG_IOCTL
+		LOG_WARNING("CDROM_IOCTL: GetAudioTracks: cdrom_fd not open");
+#endif
+		return false;
+	}
+	cdrom_tochdr toc = {};
+	if (ioctl(cdrom_fd, CDROMREADTOCHDR, &toc) != 0) {
+#ifdef DEBUG_IOCTL
+		LOG_WARNING("CDROM_IOCTL: GetAudioTracks: ioctl CDROMREADTOCHDR failed");
+#endif
+		return false;
+	}
+	cdrom_tocentry entry = {};
+	entry.cdte_track     = CDROM_LEADOUT;
+	entry.cdte_format    = CDROM_MSF;
+	if (ioctl(cdrom_fd, CDROMREADTOCENTRY, &entry) != 0) {
+#ifdef DEBUG_IOCTL
+		LOG_WARNING("CDROM_IOCTL: GetAudioTracks: ioctl CDROMREADTOCENTRY failed");
+#endif
+		return false;
+	}
+	stTrack     = toc.cdth_trk0;
+	end         = toc.cdth_trk1;
+	leadOut.min = entry.cdte_addr.msf.minute;
+	leadOut.sec = entry.cdte_addr.msf.second;
+	leadOut.fr  = entry.cdte_addr.msf.frame;
+#ifdef DEBUG_IOCTL
+	LOG_INFO("CDROM_IOCTL: GetAudioTracks => start track is %2d, last playable track is %2d, "
+	         "and lead-out MSF is %02d:%02d:%02d",
+	         stTrack,
+	         end,
+	         leadOut.min,
+	         leadOut.sec,
+	         leadOut.fr);
+#endif
+	return true;
+}
```

Comment:
A few blank lines here and there would help readability; now this function is just one big block of text.

Reply:
Fixed